src/values/AbstractValue.js

```diff
@@ -783,9 +787,10 @@ export default class AbstractValue extends Value {
 
   static generateErrorInformationForAbstractVal(val: AbstractValue): string {
     let names = [];
-    val.addSourceNamesTo(names);
+    let alreadyAdded = new Set();
+    val.addSourceNamesTo(names, alreadyAdded);
     if (names.length === 0) {
-      val.addSourceNamesTo(names);
+      val.addSourceNamesTo(names, alreadyAdded);
     }
```

Comment:
What is the original code doing here? Try adding to names, but if they come up empty, just do the same thing again?